File: Move Zeroes/MoveZero.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        i = 0
        while i < len(nums):
            if nums[i] == 0:
                del nums[i]
                nums.append(0)
            i+=1
            
        """
        Do not return anything, modify nums in-place instead.
        """
        

File: Move Zeroes/MoveZero.java
class Solution {
    public void moveZeroes(int[] nums) {
        int insertPos = 0; // Position to insert the next non-zero element

        // Move all non-zero elements to the front of the array
        for (int num : nums) {
            if (num != 0) {
                nums[insertPos++] = num;
            }
        }

        // Fill the rest of the array with zeros
        while (insertPos < nums.length) {
            nums[insertPos++] = 0;
        }
    }
}